nconnect.go

```diff
@@ -356,22 +370,35 @@ func (nc *nconnect) StartClient() error {
 	proxyHost := proxyAddr.IP.String()
 	proxyPort := uint16(proxyAddr.Port)
 
-	port, err := util.GetFreePort()
-	if err != nil {
-		return err
-	}
-
-	ssAddr := "127.0.0.1:" + strconv.Itoa(port)
-	nc.ssConfig.Client = ssAddr
 	nc.ssConfig.Socks = nc.opts.LocalSocksAddr
 
-	tunnel, err := tunnel.NewTunnel(nc.account, nc.opts.Identifier, ssAddr, remoteTunnelAddr, nc.opts.Tuna, nc.tunnelConfig)
+	var from, to []string
+	for index, remote := range remoteTunnelAddr {
+		port, err := util.GetFreePort()
+		if err != nil {
+			return err
+		}
+
+		ssAddr := "127.0.0.1:" + strconv.Itoa(port)
+		from = append(from, ssAddr)
+		to = append(to, remote)
+
+		if remoteInfo, ok := nc.remoteInfoByTunnel[remote]; ok {
+			for _, addr := range remoteInfo.LocalIP.Ipv4 {
+				nc.ssConfig.Target2Client[addr] = ssAddr
+			}
+		}
+		if index == 0 { // the first config is the default client
+			nc.ssConfig.DefaultClient = ssAddr
+		}
+	}
+	tunnels, err := tunnel.NewTunnels(nc.account, nc.opts.Identifier, from, to, nc.opts.Tuna, nc.tunnelConfig)
 	if err != nil {
 		return err
 	}
-	nc.tunnel = tunnel
+	nc.tunnels = tunnels
+	nc.ssConfig.Client = from[0]
```

Comment:
Is `ssConfig.Client` still used?

Reply:
It is kept just for compatible with ss. 
`
if flags.Client != "" { // client mode
`